tests/nlu/base/test_persistor.py

```diff
@@ -142,3 +142,27 @@ def test_retrieve_tar_archive(model, archive):
         with patch.object(persistor.Persistor, "_retrieve_tar") as f:
             persistor.Persistor().retrieve(model, "dst")
         f.assert_called_once_with(archive)
+
+
+# noinspection PyPep8Naming
+@mock_s3
+def test_retrieve_tar_archive_with_s3_namespace():
+    model = "/my/s3/project/model.tar.gz"
+    destination = "dst"
+    with patch.object(persistor.AWSPersistor, "_decompress") as decompress:
+        with patch.object(persistor.AWSPersistor, "_retrieve_tar") as retrieve:
+            persistor.AWSPersistor("rasa-test").retrieve(model, destination)
+        decompress.assert_called_once_with("model.tar.gz", destination)
+        retrieve.assert_called_once_with(model)
+
+
+# noinspection PyPep8Naming
+@mock_s3
+def test_s3_private_retrieve_tar():
```

Comment:
what exactly is being tested here? A comment would be very helpful

Reply:
Added! I included this test retroactively to bump coverage back to an acceptable level by coveralls. I'd be happy to also remove it if you think it excessive